### PR TITLE
Coverage: calling array methods with boolean as thisValue. Fixes gh-2803

### PR DESCRIPTION
**Pull request by rwaldron**

### CODE REVIEW

File: test/built-ins/Array/prototype/sort/call-with-boolean.js [hunk 1]
```diff
@@ -0,0 +1,10 @@
+// Copyright (c) 2020 Rick Waldron.  All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+
+/*---
+esid: sec-array.prototype.sort
+description: Array.prototype.sort applied to boolean primitive
+---*/
+
+assert.sameValue(Array.prototype.sort.call(true), true);
```
Comment (shvaikalesh) [inline test/built-ins/Array/prototype/sort/call-with-boolean.js:9]: Per https://github.com/tc39/ecma262/pull/1585, `? ToObject(this value)` is returned.

File: test/built-ins/Array/prototype/reverse/call-with-boolean.js [hunk 2]
```diff
@@ -0,0 +1,10 @@
+// Copyright (c) 2020 Rick Waldron.  All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+
+/*---
+esid: sec-array.prototype.reverse
+description: Array.prototype.reverse applied to boolean primitive
+---*/
+
+assert.sameValue(Array.prototype.reverse.call(true) instanceof Boolean, true);
```
Comment (shvaikalesh) [inline test/built-ins/Array/prototype/reverse/call-with-boolean.js:9]:
```suggestion
assert(Array.prototype.reverse.call(true) instanceof Boolean);
```

File: test/built-ins/Array/prototype/sort/call-with-boolean.js (hunk 1 above)
Comment (rwaldron) [reply test/built-ins/Array/prototype/sort/call-with-boolean.js:9]: I'll back this one out

Comment (mathiasbynens) [reply test/built-ins/Array/prototype/sort/call-with-boolean.js]: Filed https://bugs.chromium.org/p/v8/issues/detail?id=11362

**Pull request merged**